src/NumberTheory/QuadFormAndIsom/finite_group_actions.jl

```diff
@@ -0,0 +1,871 @@
+arguments in `kwargs` are optional arguments in the computations of such a
+group (see [`isometry_group(::ZZLat)`](@ref)).
+
+# Examples
+```jldoctest
+julia> A4 = root_lattice(:A, 4)
+Integer lattice of rank 4 and degree 4
+with gram matrix
+[ 2   -1    0    0]
+[-1    2   -1    0]
+[ 0   -1    2   -1]
+[ 0    0   -1    2]
+
+julia> A2 = lattice_in_same_ambient_space(A4, QQ[1 0 0 0; 0 1 0 0])
+Integer lattice of rank 2 and degree 4
+with gram matrix
+[ 2   -1]
+[-1    2]
+
+julia> H = pointwize_stabilizer_in_orthogonal_group(A4, A2)
+Matrix group of degree 4
+  over rational field
+
+julia> order(H)
+2
+```
+"""
+function pointwize_stabilizer_in_orthogonal_group(
+    L::ZZLat,
+    S::ZZLat;
+    kwargs...
+  )
+  @req ambient_space(L) === ambient_space(S) "Lattices are not contained in the same quadratic space"
+  return stabilizer_in_orthogonal_group(L, basis_matrix(S); kwargs...)
+end
+
+@doc raw"""
+    _stabilizer_in_diagonal_action(
+      L::ZZLat,
+      K::ZZLat,
+      N::ZZLat,
+      OK::MatrixGroup,
+      ON::MatrixGroup,
+    ) -> Vector{QQMatrix}
+
+Given a primitive extension ``K\oplus N \subseteq L`` of even lattices,
+and given two finite groups of isometries ``OK`` and ``ON`` of ``K`` and ``N``,
+respectively, return generators for the stabilizer of ``L`` in ``OK\times ON``,
+seen as a group of isometries of the ambient space of ``L``.
+"""
+function _stabilizer_in_diagonal_action(
+    L::ZZLat,
+    K::ZZLat,
+    N::ZZLat,
+    OK::MatrixGroup,
+    ON::MatrixGroup;
+    gen_fun::Function=gens,
+  )
+  # Necessary tests and can speed up kernel computations
+  @assert is_finite(OK)
+  @assert is_finite(ON)
+
+  # Need the glue map to determine isometries of OK\times ON preserving L in
+  # K^\vee\oplus N^\vee
+  phi, HKinqK, HNinqN = glue_map(L, K, N)
+  iphi = inv(phi)
+
+  gen = QQMatrix[]
+  qN = codomain(HNinqN)
+  genN = ZZMatrix[matrix(hom(qN, qN, elem_type(qN)[qN(lift(a)*matrix(g)) for a in gens(qN)])) for g in gen_fun(ON)]
+  GN = Oscar._orthogonal_group(qN, genN; check=false)
+  discN = hom(ON, GN, gens(GN); check=false)
+  stabN, _ = stabilizer(GN, HNinqN)
+  SN, resN = restrict_automorphism_group(stabN, HNinqN; check=false)
+  # Largest subgroup of ON satisfying that {id_K}\times kerN preserves L
+  kerN, _ = preimage(discN, first(kernel(resN)))
+  append!(gen, matrix.(gen_fun(kerN)))
+
+  qK = codomain(HKinqK)
+  genK = ZZMatrix[matrix(hom(qK, qK, elem_type(qK)[qK(lift(a)*matrix(g)) for a in gens(qK)])) for g in gen_fun(OK)]
+  GK = Oscar._orthogonal_group(qK, genK; check=false)
+  discK = hom(OK, GK, gens(GK); check=false)
+  stabK, _ = stabilizer(GK, HKinqK)
+  SK, resK = restrict_automorphism_group(stabK, HKinqK; check=false)
+  # Largest subgroup of OK satisfying that kerK\times {id_N} preserves L
+  kerK, _ = preimage(discK, first(kernel(resK)))
+  append!(gen, matrix.(gen_fun(kerK)))
+
+  SNphi = Oscar._orthogonal_group(domain(SK), ZZMatrix[matrix(phi * hom(g) * iphi) for g in gens(SN)])
```

Comment:
I wonder if going through the matrices is really necessary here. If it is perhaps we should reconsider our design?
```suggestion
  SNphi = Oscar._orthogonal_group(domain(SK), TorQuadModMap[phi * hom(g) * iphi for g in gens(SN)])
```

Reply:
I will have a look at that.